### PR TITLE
ISPN-7035 Spring Session 2.1 support

### PR DESCRIPTION
**Pull request by karesti**
Spring Session was not using the good version
Some API changes are included
https://issues.jboss.org/browse/ISPN-7035

### CODE REVIEW

**Comment by karesti** [comment]
remote tests are failing now in IT test module

**Comment by karesti** [comment]
All tests related to spring should be working now

**Comment by DanielFran** [comment]
@karesti Maybe you should target Tomcat Tomcat 9.0.13 instead of Tomcat 8.5?

**Comment by karesti** [comment]
@DanielFran Thanks for pointing it! We are not using that tomcat version in infinispan for now, the infinispan dependencies for spring should not bring this tomcat version to the classpath but the supported version of spring 5 instead

File: spring/spring5/spring5-remote/src/main/java/org/infinispan/spring/remote/session/RemoteApplicationPublishedBridge.java [hunk 1]
```diff
@@ -20,8 +25,11 @@
 @ClientListener
 public class RemoteApplicationPublishedBridge extends AbstractApplicationPublisherBridge {
 
-   public RemoteApplicationPublishedBridge(SpringCache eventSource) {
+   private final TaskExecutor taskExecutor;
```
Comment (galderz) [inline spring/spring5/spring5-remote/src/main/java/org/infinispan/spring/remote/session/RemoteApplicationPublishedBridge.java]: What is the reason for adding this?

**Comment by karesti** [comment]
If you do a get from the listener, there is a sort of deadlock

Le jeu. 6 déc. 2018 à 09:19, Galder Zamarreño <notifications@github.com> a
écrit :

**Pull request merged**